test/bin_test.py

```diff
@@ -32,4 +38,11 @@ class LuigidDaemonTest(server_test.ServerTestRun):
 
     @mock.patch('daemon.DaemonContext')
     def run_server(self, daemon_context):
-        luigi.cmdline.luigid(['--port', str(self._api_port), '--background', '--logdir', '.', '--pidfile', self.id() + '.pid'])
+        tempdir = tempfile.mkdtemp(prefix=self.id())
```

Comment:
hey you should probably use a context manager so you don't have to use `addCleanup`

https://docs.python.org/3/library/tempfile.html#examples

i've had some issues with tempfile before so not 100% that it works well